src/lib/pubkey/classic_mceliece/cmce_poly.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+ * Classic McEliece Polynomials
+ * Based on the public domain reference implementation by the designers
+ * (https://classic.mceliece.org/impl.html - released in Oct 2022 for NISTPQC-R4)
+ *
+ * (C) 2023 Jack Lloyd
+ *     2023,2024 Fabian Albert, Amos Treiber - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ **/
+
+#include <botan/internal/cmce_poly.h>
+#include <botan/internal/ct_utils.h>
+#include <botan/internal/stl_util.h>
+
+namespace Botan {
+
+namespace {
+
+/**
+ * @brief loads @p bytes into a GF_Elem vector in Little Endian.
+ *
+ * Replaces load_le<GF_Elem> as it cannot be compiled on Big Endian architectures
+ *
+ * @param bytes input bytes
+ * @return The vector of GF_Elem elements
+ */
+std::vector<GF_Elem> load_le_gf_vec(std::span<const uint8_t> bytes) {
+   std::vector<GF_Elem> result;
+   BufferSlicer bs(bytes);
+   result.reserve(bytes.size() / sizeof(GF_Elem));
+   for(size_t i = 0; i < bytes.size() / sizeof(GF_Elem); ++i) {
+      result.emplace_back(load_le<uint16_t>(bs.take<sizeof(GF_Elem)>()));
+   }
+   return result;
+}
+
+}  // namespace
+
+Classic_McEliece_GF Classic_McEliece_Polynomial::operator()(Classic_McEliece_GF a) const {
+   BOTAN_ASSERT(a.modulus() == coef_at(0).modulus(), "Unmatching Galois fields");
```

Comment:
Nit/FYI: The error message rendered for `BOTAN_ASSERT` (and friends) assumes that the error string is the "assertion made". I.e. it should be formulated in a positive form and will be rendered as `False assertion: '...'`.

```suggestion
   BOTAN_ASSERT(a.modulus() == coef_at(0).modulus(), "Galois fields match");
```